src/renderer/components/version-select.tsx

```diff
@@ -58,15 +58,32 @@ export function getItemIcon({ state }: RunnableVersion) {
  * Helper method: Returns the <Select /> predicate for an Electron
  * version.
  *
+ * Sorts by index of the chosen query.
+ * For example, if we take the following versions:
+ * [3.0.0, 14.3.0, 13.2.0, 12.0.0-nightly.20210301, 12.0.0-beta.3]
+ * and a search query of '3', this method would sort them into:
+ * [3.0.0, 13.2.0, 14.3.0, 12.0.0-beta.3, 12.0.0-nightly.20210301]
+ *
  * @param {string} query
- * @param {RunnableVersion} { version }
+ * @param {RunnableVersion[]} versions
  * @returns
  */
-export const filterItem: ItemPredicate<RunnableVersion> = (
+export const filterItems: ItemListPredicate<RunnableVersion> = (
   query,
-  { version },
+  versions,
 ) => {
-  return version.toLowerCase().includes(query.toLowerCase());
+  if (query === '') return versions;
+
+  const q = query.toLowerCase();
+
+  return versions
+    .filter(({ version }) => version.toLowerCase().includes(q))
+    .sort((a, b) => {
+      if (a.version.indexOf(q) > b.version.indexOf(q)) return 1;
+      if (b.version.indexOf(q) > a.version.indexOf(q)) return -1;
+
+      return 0;
+    });
```

Comment:
We can eliminate half of these searches

```suggestion
    .sort((a, b) => a.version.indexOf(q) - b.version.indexOf(q)));
```